mvnw clean install -Pdistribution on Windows deletes necessary files during clean of org.keycloak:keycloak-admin-ui

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

Running 
mvnw -Dmaven.build.cache.enabled=true clean install -Pdistribution
(on Windows system) seems to delete too much files during clean of org.keycloak:keycloak-admin-ui so that build fails and pom files of other components are missing afterwards.

### Version

999.0.0-SNAPSHOT

### Regression

- [ ] The issue is a regression

### Expected behavior

The clean of org.keycloak:keycloak-admin-ui shouldn't delete files necessary for the build process.

Using only mvnw install -Pdistribution it seems to work

### Actual behavior

./mvnw clean install -Pdistribution

...
[INFO] ----------------< org.keycloak:keycloak-js-adapter-jar >----------------
[INFO] Building Keycloak JavaScript Adapter (JAR) 999.0.0-SNAPSHOT     [20/196]
[INFO]   from adapters\oidc\js\pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- clean:3.1.0:clean (default-clean) @ keycloak-js-adapter-jar ---
[INFO] Deleting C:\Users\builduser\keycloak\js\libs\keycloak-js\dist (includes = [], excludes = [*.d.ts])
[INFO]
[INFO] --- enforcer:3.0.0-M3:enforce (enforce-java-version) @ keycloak-js-adapter-jar ---
[INFO]
[INFO] --- enforcer:3.0.0-M3:enforce (enforce-maven-version) @ keycloak-js-adapter-jar ---
[INFO]
[INFO] --- buildnumber:1.4:create (get-scm-revision) @ keycloak-js-adapter-jar ---
[INFO]
[INFO] --- frontend:1.14.2:install-node-and-pnpm (default) @ keycloak-js-adapter-jar ---
[INFO] Installing node version v18.18.2
[INFO] Copying node binary from C:\Users\builduser\.m2\repository\com\github\eirslett\node\18.18.2\node-18.18.2-win-x64.exe to C:\Users\builduser\keycloak\node\node.exe
[INFO] Installed node locally.
[INFO] Installing pnpm version 8.10.0
[INFO] Unpacking C:\Users\builduser\.m2\repository\com\github\eirslett\pnpm\8.10.0\pnpm-8.10.0.tar.gz into C:\Users\builduser\keycloak\node\node_modules
[INFO] Installed pnpm locally.
[INFO] No pnpm executable found, creating proxy script to C:\Users\builduser\keycloak\node\node_modules\pnpm\bin\pnpm.cjs.
[INFO]
[INFO] --- frontend:1.14.2:pnpm (pnpm-install) @ keycloak-js-adapter-jar ---
[INFO] Running 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' in C:\Users\builduser\keycloak\js\libs\keycloak-js
[INFO] Scope: all 8 workspace projects
[INFO] ../..                                    | Progress: resolved 1, reused 0, downloaded 0, added 0
[INFO] ../..                                    | +988 ++++++++++++++++++++++++++++++++
[INFO] ../..                                    | Progress: resolved 988, reused 25, downloaded 0, added 0
[INFO] ../..                                    | Progress: resolved 988, reused 51, downloaded 0, added 0
[INFO] ../..                                    | Progress: resolved 988, reused 212, downloaded 0, added 0
[INFO] ../..                                    | Progress: resolved 988, reused 347, downloaded 0, added 0
[INFO] ../..                                    | Progress: resolved 988, reused 369, downloaded 0, added 0
[INFO] ../..                                    | Progress: resolved 988, reused 547, downloaded 0, added 0
[INFO] ../..                                    | Progress: resolved 988, reused 769, downloaded 0, added 0
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 0
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 46
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 122
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 230
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 307
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 345
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 351
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 352
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 366
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 368
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 388
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 392
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 429
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 469
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 528
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 613
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 670
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 750
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 770
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 842
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 851
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 852
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 921
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 930
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 979
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 987
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 988
[INFO] ../..                                    | Progress: resolved 988, reused 988, downloaded 0, added 988, done
[INFO]
[INFO] dependencies:
[INFO] + js-sha256 0.11.0
[INFO] + jwt-decode 4.0.0
[INFO]
[INFO] devDependencies:
[INFO] + @rollup/plugin-commonjs 25.0.7
[INFO] + @rollup/plugin-inject 5.0.5
[INFO] + @rollup/plugin-node-resolve 15.2.3
[INFO] + @rollup/plugin-terser 0.4.4
[INFO] + @rollup/plugin-typescript 11.1.6
[INFO] + es6-promise 4.2.8
[INFO] + rollup 4.12.0
[INFO] + shx 0.3.4
[INFO]
[INFO] Done in 49.4s
[INFO]
[INFO] --- frontend:1.14.2:pnpm (run-build) @ keycloak-js-adapter-jar ---
[INFO] Running 'pnpm run build' in C:\Users\builduser\keycloak\js\libs\keycloak-js
[INFO]
[INFO] > keycloak-js@999.0.0-SNAPSHOT build C:\Users\builduser\keycloak\js\libs\keycloak-js
[INFO] > wireit
[INFO]
[INFO] Analyzing
[INFO]   0% [0 / 1] [1 running] build
[INFO]
[INFO] src\keycloak.js ? dist\keycloak.mjs...
[INFO] created dist\keycloak.mjs in 232ms
[INFO]
[INFO] src\keycloak.js ? dist\keycloak.js, dist\keycloak.min.js...
[INFO] (!) Missing shims for Node.js built-ins
[INFO] Creating a browser bundle that depends on "crypto" and "buffer". You might need to include https://github.com/FredKSchott/rollup-plugin-polyfill-node
[INFO] (!) Missing shims for Node.js built-ins
[INFO] Creating a browser bundle that depends on "crypto" and "buffer". You might need to include https://github.com/FredKSchott/rollup-plugin-polyfill-node
[INFO] (!) Missing global variable names
[INFO] https://rollupjs.org/configuration-options/#output-globals
[INFO] Use "output.globals" to specify browser global variable names corresponding to external modules:
[INFO] crypto (guessing "require$$0")
[INFO] buffer (guessing "require$$1")
[INFO] crypto (guessing "require$$0")
[INFO] buffer (guessing "require$$1")
[INFO] (!) Plugin node-resolve: preferring built-in module 'buffer' over local alternative at 'C:\Users\builduser\keycloak\js\node_modules\.pnpm\buffer@5.7.1\node_modules\buffer\index.js', pass 'preferBuiltins: false' to disable this behavior or 'preferBuiltins: true' to disable this warning
[INFO] created dist\keycloak.js, dist\keycloak.min.js in 2.4s
[INFO]
[INFO] src\keycloak-authz.js ? dist\keycloak-authz.mjs...
[INFO] created dist\keycloak-authz.mjs in 35ms
[INFO]
[INFO] src\keycloak-authz.js ? dist\keycloak-authz.js, dist\keycloak-authz.min.js...
[INFO] created dist\keycloak-authz.js, dist\keycloak-authz.min.js in 833ms
[INFO]
[INFO] > keycloak-js@999.0.0-SNAPSHOT duplicate-types C:\Users\builduser\keycloak\js\libs\keycloak-js
[INFO] > shx cp dist/keycloak.d.ts dist/keycloak.d.mts && shx cp dist/keycloak-authz.d.ts dist/keycloak-authz.d.mts
[INFO]
[INFO] ? Ran 1 script and skipped 0 in 10,2s.
[INFO]
[INFO] --- resources:3.2.0:resources (default-resources) @ keycloak-js-adapter-jar ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] Copying 4 resources
[INFO]
[INFO] --- compiler:3.8.1:compile (default-compile) @ keycloak-js-adapter-jar ---
[INFO] No sources to compile
[INFO]
[INFO] --- resources:3.2.0:testResources (default-testResources) @ keycloak-js-adapter-jar ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] skip non existing resourceDirectory C:\Users\builduser\keycloak\adapters\oidc\js\src\test\resources
[INFO]
[INFO] --- compiler:3.8.1:testCompile (default-testCompile) @ keycloak-js-adapter-jar ---
[INFO] No sources to compile
[INFO]
[INFO] --- surefire:3.0.0-M7:test (default-test) @ keycloak-js-adapter-jar ---
[INFO] No tests to run.
[INFO]
[INFO] --- jar:3.1.2:jar (default-jar) @ keycloak-js-adapter-jar ---
[INFO] Building jar: C:\Users\builduser\keycloak\adapters\oidc\js\target\keycloak-js-adapter-jar-999.0.0-SNAPSHOT.jar
[INFO]
[INFO] --- source:3.1.0:jar-no-fork (attach-sources) @ keycloak-js-adapter-jar ---
[INFO] Building jar: C:\Users\builduser\keycloak\adapters\oidc\js\target\keycloak-js-adapter-jar-999.0.0-SNAPSHOT-sources.jar
[INFO]
[INFO] --- install:2.5.2:install (default-install) @ keycloak-js-adapter-jar ---
[INFO] Installing C:\Users\builduser\keycloak\adapters\oidc\js\target\keycloak-js-adapter-jar-999.0.0-SNAPSHOT.jar to C:\Users\builduser\.m2\repository\org\keycloak\keycloak-js-adapter-jar\999.0.0-SNAPSHOT\keycloak-js-adapter-jar-999.0.0-SNAPSHOT.jar
[INFO] Installing C:\Users\builduser\keycloak\adapters\oidc\js\pom.xml to C:\Users\builduser\.m2\repository\org\keycloak\keycloak-js-adapter-jar\999.0.0-SNAPSHOT\keycloak-js-adapter-jar-999.0.0-SNAPSHOT.pom
[INFO] Installing C:\Users\builduser\keycloak\adapters\oidc\js\target\keycloak-js-adapter-jar-999.0.0-SNAPSHOT-sources.jar to C:\Users\builduser\.m2\repository\org\keycloak\keycloak-js-adapter-jar\999.0.0-SNAPSHOT\keycloak-js-adapter-jar-999.0.0-SNAPSHOT-sources.jar
[INFO]
[INFO] ------------------< org.keycloak:keycloak-js-parent >-------------------
[INFO] Building Keycloak JavaScript Parent 999.0.0-SNAPSHOT            [21/196]
[INFO]   from js\pom.xml
[INFO] --------------------------------[ pom ]---------------------------------
[INFO]
[INFO] --- clean:3.1.0:clean (default-clean) @ keycloak-js-parent ---
[INFO] Deleting C:\Users\builduser\keycloak\js (includes = [**/.wireit/**, **/node_modules/**], excludes = [])
[INFO] Deleting C:\Users\builduser\keycloak\js\apps\keycloak-server (includes = [server/**], excludes = [])
[INFO]
[INFO] --- enforcer:3.0.0-M3:enforce (enforce-java-version) @ keycloak-js-parent ---
[INFO]
[INFO] --- enforcer:3.0.0-M3:enforce (enforce-maven-version) @ keycloak-js-parent ---
[INFO]
[INFO] --- buildnumber:1.4:create (get-scm-revision) @ keycloak-js-parent ---
[INFO]
[INFO] --- frontend:1.14.2:install-node-and-pnpm (default) @ keycloak-js-parent ---
[INFO] Node v18.18.2 is already installed.
[INFO] PNPM 8.10.0 is already installed.
[INFO] Existing pnpm executable found, skipping linking.
[INFO]
[INFO] --- frontend:1.14.2:pnpm (pnpm-install) @ keycloak-js-parent ---
[INFO] Running 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' in C:\Users\builduser\keycloak\js
[INFO] Scope: all 4 workspace projects
[INFO] Lockfile is up to date, resolution step is skipped
[INFO] Progress: resolved 1, reused 0, downloaded 0, added 0
[INFO] Packages: +988
[INFO] ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
[INFO] Progress: resolved 988, reused 20, downloaded 0, added 0
[INFO] Progress: resolved 988, reused 42, downloaded 0, added 0
[INFO] Progress: resolved 988, reused 221, downloaded 0, added 0
[INFO] Progress: resolved 988, reused 347, downloaded 0, added 0
[INFO] Progress: resolved 988, reused 362, downloaded 0, added 0
[INFO] Progress: resolved 988, reused 524, downloaded 0, added 0
[INFO] Progress: resolved 988, reused 735, downloaded 0, added 0
[INFO] Progress: resolved 988, reused 908, downloaded 0, added 0
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 19
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 100
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 221
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 303
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 339
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 351
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 352
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 366
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 368
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 388
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 402
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 433
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 521
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 556
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 663
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 752
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 786
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 856
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 857
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 860
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 917
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 959
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 986
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 987
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 988
[INFO] Progress: resolved 988, reused 988, downloaded 0, added 988, done
[INFO]
[INFO] devDependencies:
[INFO] + @types/node 20.11.20
[INFO] + @typescript-eslint/eslint-plugin 7.0.0
[INFO] + @typescript-eslint/parser 6.21.0
[INFO] + eslint 8.56.0
[INFO] + eslint-config-prettier 9.1.0
[INFO] + eslint-import-resolver-typescript 3.6.1
[INFO] + eslint-plugin-cypress 2.15.1
[INFO] + eslint-plugin-import 2.29.1
[INFO] + eslint-plugin-lodash 7.4.0
[INFO] + eslint-plugin-mocha 10.3.0
[INFO] + eslint-plugin-prettier 5.1.3
[INFO] + eslint-plugin-react 7.33.2
[INFO] + eslint-plugin-react-hooks 4.6.0
[INFO] + husky 9.0.11
[INFO] + lint-staged 15.2.2
[INFO] + prettier 3.2.5
[INFO] + tslib 2.6.2
[INFO] + typescript 5.3.3
[INFO] + wireit 0.14.4
[INFO]
[INFO] Done in 47.8s
[INFO]
[INFO] --- source:3.1.0:jar-no-fork (attach-sources) @ keycloak-js-parent ---
[INFO]
[INFO] --- install:2.5.2:install (default-install) @ keycloak-js-parent ---
[INFO] Installing C:\Users\builduser\keycloak\js\pom.xml to C:\Users\builduser\.m2\repository\org\keycloak\keycloak-js-parent\999.0.0-SNAPSHOT\keycloak-js-parent-999.0.0-SNAPSHOT.pom
[INFO]
[INFO] -------------------< org.keycloak:keycloak-admin-ui >-------------------
[INFO] Building Keycloak Admin UI 999.0.0-SNAPSHOT                     [22/196]
[INFO]   from js\apps\admin-ui\pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- clean:3.1.0:clean (default-clean) @ keycloak-admin-ui ---
[INFO] Deleting C:\Users\builduser\keycloak\js\apps\admin-ui (includes = [**/.wireit/**, **/node_modules/**], excludes = [])
[INFO]
[INFO] --- enforcer:3.0.0-M3:enforce (enforce-java-version) @ keycloak-admin-ui ---
[INFO]
[INFO] --- enforcer:3.0.0-M3:enforce (enforce-maven-version) @ keycloak-admin-ui ---
[INFO]
[INFO] --- buildnumber:1.4:create (get-scm-revision) @ keycloak-admin-ui ---
[INFO]
[INFO] --- frontend:1.14.2:install-node-and-pnpm (default) @ keycloak-admin-ui ---
[INFO] Node v18.18.2 is already installed.
[INFO] PNPM 8.10.0 is already installed.
[INFO] Existing pnpm executable found, skipping linking.
[INFO]
[INFO] --- frontend:1.14.2:pnpm (pnpm-install) @ keycloak-admin-ui ---
[INFO] Running 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' in C:\Users\builduser\keycloak\js\apps\admin-ui
[INFO] Scope: all 4 workspace projects
[INFO] ../..                                    | Progress: resolved 1, reused 0, downloaded 0, added 0
[INFO] ../..                                    | +988 ++++++++++++++++++++++++++++++++
[INFO] ../..                                    | Progress: resolved 1, reused 1, downloaded 0, added 0
[INFO] ../..                                    | Progress: resolved 1, reused 1, downloaded 0, added 1, done
[INFO] ?ENOENT? ENOENT: no such file or directory, open 'C:\Users\builduser\keycloak\js\node_modules\.pnpm\vite@5.1.4_@types+node@20.11.20_lightningcss@1.23.0\node_modules\vite\package.json'
[INFO]
[INFO] pnpm: ENOENT: no such file or directory, open 'C:\Users\builduser\keycloak\js\node_modules\.pnpm\vite@5.1.4_@types+node@20.11.20_lightningcss@1.23.0\node_modules\vite\package.json'
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Keycloak 999.0.0-SNAPSHOT:
[INFO]
[INFO] Keycloak BOM Parent ................................ SUCCESS [  2.133 s]
[INFO] Keycloak BOM for adapters .......................... SUCCESS [  0.047 s]
[INFO] Keycloak BOM for server extensions ................. SUCCESS [  0.037 s]
[INFO] Keycloak BOM utilities for the quickstarts ......... SUCCESS [  0.032 s]
[INFO] Keycloak ........................................... SUCCESS [  0.100 s]
[INFO] Keycloak Common .................................... SUCCESS [ 11.012 s]
[INFO] Keycloak Core ...................................... SUCCESS [ 12.674 s]
[INFO] Keycloak Server SPI ................................ SUCCESS [  8.173 s]
[INFO] Keycloak Server Private SPI ........................ SUCCESS [ 14.319 s]
[INFO] Keycloak Crypto Parent ............................. SUCCESS [  0.047 s]
[INFO] Keycloak Crypto Default ............................ SUCCESS [ 10.721 s]
[INFO] Keycloak SAML Core Public API ...................... SUCCESS [  5.076 s]
[INFO] Keycloak SAML Core ................................. SUCCESS [ 12.348 s]
[INFO] Keycloak Model Parent .............................. SUCCESS [  0.037 s]
[INFO] Keycloak Database Support .......................... SUCCESS [  1.366 s]
[INFO] Keycloak Database Support - Private classes ........ SUCCESS [  4.261 s]
[INFO] Keycloak REST Services ............................. SUCCESS [ 43.910 s]
[INFO] Keycloak Crypto FIPS 140-2 Integration ............. SUCCESS [ 11.566 s]
[INFO] Keycloak Crypto Wildfly Elytron Integration ........ SUCCESS [  9.257 s]
[INFO] Keycloak JavaScript Adapter (JAR) .................. SUCCESS [01:04 min]
[INFO] Keycloak JavaScript Parent ......................... SUCCESS [02:21 min]
[INFO] Keycloak Admin UI .................................. FAILURE [ 34.970 s]
[INFO] Keycloak Account UI ................................ SKIPPED
[INFO] Keycloak Themes .................................... SKIPPED
[INFO] Keycloak Dependencies Parent ....................... SKIPPED
[INFO] Keycloak Dependencies Server Min ................... SKIPPED
[INFO] Keycloak Model JPA ................................. SKIPPED
[INFO] Keycloak Kerberos Federation ....................... SKIPPED
[INFO] Keycloak LDAP UserStoreProvider .................... SKIPPED
[INFO] Keycloak Database Support - REST services .......... SKIPPED
[INFO] Keycloak Model Infinispan .......................... SKIPPED
[INFO] Keycloak SSSD Federation ........................... SKIPPED
[INFO] Keycloak Authz: Parent ............................. SKIPPED
[INFO] Keycloak AuthZ: Policy Provider Parent ............. SKIPPED
[INFO] Keycloak AuthZ: Common Policy Providers ............ SKIPPED
[INFO] Keycloak Dependencies Server All ................... SKIPPED
[INFO] Keycloak Quarkus Parent ............................ SKIPPED
[INFO] Keycloak Configuration API ......................... SKIPPED
[INFO] Keycloak Administration UI ......................... SKIPPED
[INFO] Admin UI REST extensions ........................... SKIPPED
[INFO] Keycloak Quarkus Server Extension .................. SKIPPED
[INFO] Keycloak Quarkus Server Deployment ................. SKIPPED
[INFO] Keycloak Docs Parent ............................... SKIPPED
[INFO] Keycloak Guides Maven Plugin ....................... SKIPPED
[INFO] Keycloak Guides .................................... SKIPPED
[INFO] Keycloak Federation ................................ SKIPPED
[INFO] Keycloak Legacy Database Support ................... SKIPPED
[INFO] Keycloak Util Embedded LDAP ........................ SKIPPED
[INFO] Keycloak Util Parent ............................... SKIPPED
[INFO] Keycloak Integration ............................... SKIPPED
[INFO] Keycloak Admin REST Client JavaEE .................. SKIPPED
[INFO] Keycloak Admin REST Client ......................... SKIPPED
[INFO] Keycloak Client Registration API ................... SKIPPED
[INFO] Keycloak Client CLI ................................ SKIPPED
[INFO] Keycloak Client Registration CLI ................... SKIPPED
[INFO] Keycloak Admin CLI ................................. SKIPPED
[INFO] Keycloak Client CLI Distribution ................... SKIPPED
[INFO] Keycloak Adapter SPI ............................... SKIPPED
[INFO] Keycloak Tomcat Adapter SPI ........................ SKIPPED
[INFO] Keycloak Undertow Integration SPI .................. SKIPPED
[INFO] Keycloak Servlet Integration ....................... SKIPPED
[INFO] Keycloak Jakarta Servlet Integration ............... SKIPPED
[INFO] Common JBoss/Wildfly Core Classes .................. SKIPPED
[INFO] Keycloak Jetty Adapter SPI ......................... SKIPPED
[INFO] Keycloak Client Adapter SPI Modules ................ SKIPPED
[INFO] Keycloak SAML Client Adapter Public API ............ SKIPPED
[INFO] Keycloak SAML Client Adapter Core .................. SKIPPED
[INFO] Keycloak SAML Client Adapter Core Jakarta .......... SKIPPED
[INFO] Keycloak Jetty Core SAML Integration ............... SKIPPED
[INFO] Keycloak Jetty 9.4.x SAML Integration .............. SKIPPED
[INFO] Keycloak SAML Jetty Integration .................... SKIPPED
[INFO] Keycloak Undertow SAML Adapter ..................... SKIPPED
[INFO] Keycloak SAML Tomcat Integration ................... SKIPPED
[INFO] Keycloak Tomcat Core SAML Integration .............. SKIPPED
[INFO] Keycloak Tomcat SAML Integration ................... SKIPPED
[INFO] Keycloak Authz: Client API ......................... SKIPPED
[INFO] Keycloak Authz: Policy Enforcer .................... SKIPPED
[INFO] Keycloak Adapter Core .............................. SKIPPED
[INFO] Keycloak WildFly Elytron SAML Adapter .............. SKIPPED
[INFO] Keycloak Wildfly SAML Adapter Subsystem ............ SKIPPED
[INFO] Keycloak WildFly Elytron Jakarta SAML Adapter ...... SKIPPED
[INFO] Keycloak Wildfly Jakarta SAML Adapter Subsystem .... SKIPPED
[INFO] Keycloak SAML Wildfly Integration .................. SKIPPED
[INFO] Keycloak SAML Servlet Filter ....................... SKIPPED
[INFO] Keycloak SAML Jakarta Servlet Filter ............... SKIPPED
[INFO] Keycloak SAML Client Adapter Modules ............... SKIPPED
[INFO] Keycloak Installed Application ..................... SKIPPED
[INFO] Keycloak JAX-RS OAuth Client ....................... SKIPPED
[INFO] Keycloak Jetty Core Integration .................... SKIPPED
[INFO] Keycloak Jetty 9.4.x Integration ................... SKIPPED
[INFO] Keycloak Jetty Integration ......................... SKIPPED
[INFO] Keycloak Servlet Filter Adapter Integration ........ SKIPPED
[INFO] Keycloak Servlet Filter Adapter Integration ........ SKIPPED
[INFO] Keycloak Tomcat Integration ........................ SKIPPED
[INFO] Keycloak Tomcat Core Integration ................... SKIPPED
[INFO] Keycloak Tomcat Integration ........................ SKIPPED
[INFO] Keycloak Undertow Integration ...................... SKIPPED
[INFO] spring-boot-container-bundle ....................... SKIPPED
[INFO] Keycloak Spring Security Integration ............... SKIPPED
[INFO] Keycloak Spring Boot Adapter Core .................. SKIPPED
[INFO] Keycloak Spring Boot 2 Integration ................. SKIPPED
[INFO] Keycloak Wildfly Elytron OIDC Adapter .............. SKIPPED
[INFO] Keycloak Wildfly Adapter Subsystem ................. SKIPPED
[INFO] Keycloak WildFly Integration ....................... SKIPPED
[INFO] Keycloak OIDC Client Adapter Modules ............... SKIPPED
[INFO] Keycloak Adapters .................................. SKIPPED
[INFO] Keycloak Misc ...................................... SKIPPED
[INFO] Keycloak :: Spring :: Boot ......................... SKIPPED
[INFO] Keycloak :: Spring :: Boot :: Default ::  Starter .. SKIPPED
[INFO] keycloak-test-helper ............................... SKIPPED
[INFO] Keycloak JavaScript Admin Client ................... SKIPPED
[INFO] Keycloak JavaScript Adapter ........................ SKIPPED
[INFO] Keycloak Quarkus Server App ........................ SKIPPED
[INFO] Keycloak Quarkus Server Distribution ............... SKIPPED
[INFO] Keycloak Quarkus Test Parent ....................... SKIPPED
[INFO] Keycloak Quarkus Server JUnit 5 Internal Test Framework SKIPPED
[INFO] Keycloak Quarkus Server Integration tests .......... SKIPPED
[INFO] Keycloak TestSuite ................................. SKIPPED
[INFO] DB Allocator Plugin ................................ SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite .......... SKIPPED
[INFO] Test apps .......................................... SKIPPED
[INFO] Test apps distribution ............................. SKIPPED
[INFO] Keycloak Authz Tests: Hello World Example .......... SKIPPED
[INFO] Keycloak Authz: Servlet Authorization Test ......... SKIPPED
[INFO] Keycloak Authz: Simple Servlet App with Policy Enforcer SKIPPED
[INFO] integration-arquillian-test-apps-servlets .......... SKIPPED
[INFO] integration-arquillian-test-apps-servlets-jakarta .. SKIPPED
[INFO] Keycloak Test App Profile JEE ...................... SKIPPED
[INFO] integration-arquillian-test-apps-cors-parent ....... SKIPPED
[INFO] Angular Product Portal JS .......................... SKIPPED
[INFO] JAX-RS Database Service Using OAuth Bearer Tokens .. SKIPPED
[INFO] spring-boot-adapter-app ............................ SKIPPED
[INFO] Fuse Test Applications ............................. SKIPPED
[INFO] Customer Portal - Secured in Karaf/Fuse ............ SKIPPED
[INFO] CXF JAXRS Example - Secured in Karaf/Fuse .......... SKIPPED
[INFO] CXF JAXRS Example - Secured in Karaf/Fuse 7 on Undertow SKIPPED
[INFO] CXF JAXWS Example - Secured in Karaf/Fuse .......... SKIPPED
[INFO] CXF JAXWS Example - Secured in Karaf/Fuse 7 on Undertow SKIPPED
[INFO] Product Portal - Secured in Karaf/Fuse ............. SKIPPED
[INFO] Product Portal - Secured in Karaf/Fuse 7 on Undertow SKIPPED
[INFO] Camel endpoint example - Secured in Karaf/Fuse ..... SKIPPED
[INFO] Camel endpoint example - Secured in Karaf/Fuse 7.0 on Undertow SKIPPED
[INFO] Keycloak Fuse Example - Features ................... SKIPPED
[INFO] Keycloak Examples - External Config ................ SKIPPED
[INFO] Servers ............................................ SKIPPED
[INFO] Testsuite Adapter SPI .............................. SKIPPED
[INFO] Adapter Undertow SPI (JakartaEE) ................... SKIPPED
[INFO] Undertow OIDC Adapter (JakartaEE) .................. SKIPPED
[INFO] Undertow SAML Adapter (JakartaEE) .................. SKIPPED
[INFO] Auth Server ........................................ SKIPPED
[INFO] Auth Server Services ............................... SKIPPED
[INFO] Auth Server Services - Testsuite Providers Deployment SKIPPED
[INFO] Auth Server Services - Testsuite Providers ......... SKIPPED
[INFO] Keycloak TestSuite Utils ........................... SKIPPED
[INFO] Test Util .......................................... SKIPPED
[INFO] Auth Server - Undertow ............................. SKIPPED
[INFO] Auth Server - Quarkus .............................. SKIPPED
[INFO] App Server ......................................... SKIPPED
[INFO] App Server - SPI ................................... SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server - Infinispan SKIPPED
[INFO] Keycloak Arquillian Integration TestSuite - Cache Server - Legacy SKIPPED
[INFO] Tests .............................................. SKIPPED
[INFO] Base TestSuite ..................................... SKIPPED
[INFO] Other Tests Modules ................................ SKIPPED
[INFO] Tests for logical storage layer .................... SKIPPED
[INFO] Keycloak Examples .................................. SKIPPED
[INFO] Keycloak Examples - Admin Client ................... SKIPPED
[INFO] Provider Examples .................................. SKIPPED
[INFO] REST Example ....................................... SKIPPED
[INFO] Authenticator Example .............................. SKIPPED
[INFO] JS Console ......................................... SKIPPED
[INFO] Keycloak Examples - Kerberos Credential Delegation . SKIPPED
[INFO] Themes Examples .................................... SKIPPED
[INFO] SAML Examples ...................................... SKIPPED
[INFO] Keycloak SAML Adapter as a Servlet Filter .......... SKIPPED
[INFO] LDAP Demo Application .............................. SKIPPED
[INFO] Distribution Parent ................................ SKIPPED
[INFO] Feature Pack Builds ................................ SKIPPED
[INFO] Keycloak Distribution Licenses Common .............. SKIPPED
[INFO] Keycloak Distribution Maven Plugins Parent ......... SKIPPED
[INFO] Keycloak Licenses Processor Maven Plugin ........... SKIPPED
[INFO] Keycloak Feature Pack: Adapter ..................... SKIPPED
[INFO] Adapters Distribution Parent ....................... SKIPPED
[INFO] Keycloak Adapter Overlay Distribution .............. SKIPPED
[INFO] Keycloak Tomcat Adapter Distro ..................... SKIPPED
[INFO] Keycloak SAML Wildfly Modules ...................... SKIPPED
[INFO] Keycloak SAML Wildfly Jakarta Modules .............. SKIPPED
[INFO] Keycloak SAML Wildfly Adapter Jakarta Distro ....... SKIPPED
[INFO] Keycloak SAML Wildfly Adapter Distro ............... SKIPPED
[INFO] Keycloak Wildfly SAML Adapter ...................... SKIPPED
[INFO] Keycloak SAML Tomcat Adapter Distro ................ SKIPPED
[INFO] SAML Adapters Distribution Parent .................. SKIPPED
[INFO] Galleon Feature Pack Builds ........................ SKIPPED
[INFO] Keycloak Galleon Feature Pack: SAML Adapter ........ SKIPPED
[INFO] Keycloak Galleon Feature Pack: SAML Adapter Layer Metadata Tests SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  06:34 min
[INFO] Finished at: 2024-02-25T00:47:07+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.14.2:pnpm (pnpm-install) on project keycloak-admin-ui: Failed to run task: 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :keycloak-admin-ui

mvnw clean install -Pdistribution -rf :keycloak-admin-ui
Picked up JAVA_TOOL_OPTIONS: "-Dfile.encoding=UTF8"
[INFO] Loading cache configuration from C:\Users\builduser\keycloak\.mvn\maven-build-cache-config.xml
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] Child module C:\Users\builduser\keycloak\js\libs\keycloak-admin-client\pom.xml of C:\Users\builduser\keycloak\js\pom.xml does not exist @
[ERROR] Child module C:\Users\builduser\keycloak\js\libs\keycloak-js\pom.xml of C:\Users\builduser\keycloak\js\pom.xml does not exist @
 @
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]
[ERROR]   The project org.keycloak:keycloak-js-parent:999.0.0-SNAPSHOT (C:\Users\builduser\keycloak\js\pom.xml) has 2 errors
[ERROR]     Child module C:\Users\builduser\keycloak\js\libs\keycloak-admin-client\pom.xml of C:\Users\builduser\keycloak\js\pom.xml does not exist
[ERROR]     Child module C:\Users\builduser\keycloak\js\libs\keycloak-js\pom.xml of C:\Users\builduser\keycloak\js\pom.xml does not exist
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException

### How to Reproduce?

On a Windows system
Install JDK 17.0.10
Install Git 2.44.0
Start Git CMD
git clone https://github.com/keycloak/keycloak.git
cd keycloak
echo Change file encoding to UTF8, otherwise failing tests break the build earlier
set JAVA_TOOL_OPTIONS="-Dfile.encoding=UTF8"
mvnw clean install -Pdistribution

### Anything else?

_No response_

--- a/js/pom.xml
+++ b/js/pom.xml
@@ -27,6 +27,8 @@
         <maven.build.cache.input.1>src</maven.build.cache.input.1>
         <!-- The child projects will package from the 'dist' folder, which is listed as a resource, but isn't a source folder -->
         <maven.build.cache.exclude.value.1>${project.basedir}/dist</maven.build.cache.exclude.value.1>
+        <!-- The clean step is skipped on Windows -->
+        <js.skip.clean>false</js.skip.clean>
     </properties>
 
     <build>
@@ -56,6 +58,8 @@
                 <artifactId>maven-clean-plugin</artifactId>
                 <configuration>
                     <followSymLinks>false</followSymLinks>
+                    <!-- Skip this step if on windows -->
+                    <skip>${js.skip.clean}</skip>
                     <filesets>
                         <fileset>
                             <directory>${basedir}</directory>
@@ -76,4 +80,18 @@
             </plugin>
         </plugins>
     </build>
+
+    <profiles>
+        <profile>
+            <id>clean-when-not-on-windows</id>
+            <activation>
+                <os>
+                    <family>windows</family>
+                </os>
+            </activation>
+            <properties>
+                <js.skip.clean>true</js.skip.clean>
+            </properties>
+        </profile>
+    </profiles>
 </project>

